src/install/PackageManager/install_with_manager.zig

```diff
@@ -989,6 +994,740 @@ fn printBlockedPackagesInfo(summary: *const PackageInstall.Summary, global: bool
+
+        try writer.print(
+            \\  {{
+            \\    "name": {},
+            \\    "version": "{s}",
+            \\    "requestedRange": {},
+            \\    "tarball": {}
+            \\  }}
+        , .{ bun.fmt.formatJSONStringUTF8(pkg_name.slice(string_buf), .{}), pkg_res.value.npm.version.fmt(string_buf), bun.fmt.formatJSONStringUTF8(dep_version.literal.slice(string_buf), .{}), bun.fmt.formatJSONStringUTF8(pkg_res.value.npm.url.slice(string_buf), .{}) });
+
+        first = false;
+
+        // then go through it's dependencies and queue them up if
+        // valid and first time we've seen them
+        const pkg_deps = pkg_dependencies[pkg_id];
+
+        for (pkg_deps.begin()..pkg_deps.end()) |_next_dep_id| {
+            const next_dep_id: DependencyID = @intCast(_next_dep_id);
+
+            const next_pkg_id = manager.lockfile.buffers.resolutions.items[next_dep_id];
+            if (next_pkg_id == invalid_package_id) {
+                continue;
+            }
+
+            const next_pkg_res = pkg_resolutions[next_pkg_id];
+            if (next_pkg_res.tag != .npm) {
+                continue;
+            }
+
+            if ((try pkg_dedupe.getOrPut(next_pkg_id)).found_existing) {
+                continue;
+            }
+
+            var extended_pkg_path = std.ArrayList(PackageID).init(manager.allocator);
+            try extended_pkg_path.appendSlice(item.pkg_path.items);
+            try extended_pkg_path.append(next_pkg_id);
+
+            var extended_dep_path = std.ArrayList(DependencyID).init(manager.allocator);
+            try extended_dep_path.appendSlice(item.dep_path.items);
+            try extended_dep_path.append(next_dep_id);
+
+            try ids_queue.writeItem(.{
+                .pkg_id = next_pkg_id,
+                .dep_id = next_dep_id,
+                .pkg_path = extended_pkg_path,
+                .dep_path = extended_dep_path,
+            });
+        }
+    }
+
+    try writer.writeAll("\n]");
+
+    var code_buf = std.ArrayList(u8).init(manager.allocator);
+    defer code_buf.deinit();
+    var code_writer = code_buf.writer();
+
+    try code_writer.print(
+        \\try {{
+        \\  const {{provider}} = await import('{s}');
+        \\  const packages = {s};
+        \\
+        \\  if (provider.version !== '1') {{
+        \\    throw new Error('Security provider must be version 1');
+        \\  }}
+        \\
+        \\  if (typeof provider.scan !== 'function') {{
+        \\    throw new Error('provider.scan is not a function');
+        \\  }}
+        \\
+        \\  const result = await provider.scan({{packages:packages}});
+        \\
+        \\  if (!Array.isArray(result)) {{
+        \\    throw new Error('Security provider must return an array of advisories');
+        \\  }}
+        \\
+        \\  const fs = require('fs');
+        \\  const data = JSON.stringify({{advisories: result}});
+        \\  fs.writeSync(3, data);
```

Comment:
this needs to track how many bytes were written in each write call and how many remain, and loop until it writes everything or loop until the number it returns (written) is 0
```suggestion
       \\  for (let remaining = data; remaining.length > 0;)  {{
        \\  const written = fs.writeSync(3, remaining);
        \\  if (written === 0) process.exit(1);
        \\ remaining = remaining.slice(written);
        \\}}
```